toast-tk-rest-utils/src/main/java/io/toast/tk/core/rest/RestUtils.java

```diff
@@ -273,4 +304,58 @@ public static String downloadRepositoryAsWiki(String apiKey) {
 	public static void unRegisterAgent(String hostName) {
 		// NO-OP
 	}
+
+	private static PoolingHttpClientConnectionManager setHttpConnectionManager() throws NoSuchAlgorithmException, KeyStoreException, KeyManagementException {
+		// This connection manager must be used if more than one thread will
+	       // be using the HttpClient.
+			SSLContextBuilder builder = SSLContexts.custom();
+			builder.loadTrustMaterial(null, new TrustStrategy() {
+			    @Override
+			    public boolean isTrusted(X509Certificate[] chain, String authType)
+			            throws CertificateException {
+			        return true;
+			    }
+			});
+			
+			@SuppressWarnings("deprecation")
+			SSLConnectionSocketFactory sslsf = new SSLConnectionSocketFactory(
+	                builder.build(),SSLConnectionSocketFactory.ALLOW_ALL_HOSTNAME_VERIFIER);
+			@SuppressWarnings("unused")
+			PlainConnectionSocketFactory pcsf = new PlainConnectionSocketFactory();
+			
+			Registry<ConnectionSocketFactory> socketFactoryRegistry = RegistryBuilder
+			        .<ConnectionSocketFactory> create().register("https", sslsf)
+			        .register("http", PlainConnectionSocketFactory.getSocketFactory())
+			        .build();
+
+	        PoolingHttpClientConnectionManager cm = new PoolingHttpClientConnectionManager(socketFactoryRegistry);
+	        cm.setMaxTotal(HTTP_CLIENT_MAX_POOL_SIZE);
+	        //cm.setDefaultSocketConfig( SocketConfig.custom().setSoKeepAlive( true ).setSoReuseAddress( true ).setSoTimeout( 3000 ).build() 
+	        //cm.setValidateAfterInactivity(1); // essai pour resoudre java.net.SocketException: Software caused connection abort: recv failed
+
+	        @SuppressWarnings("unused")
+			ConnectionKeepAliveStrategy myStrategy = new ConnectionKeepAliveStrategy() {

```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Make this anonymous inner class a lambda [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1604)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "myStrategy". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1854)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unused "myStrategy" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1481)